### PR TITLE
remove conditional rendering cases from color legend

### PR DESCRIPTION
**Pull request by seve**
The color legend component was overly protective of showing/rendering itself when those cases were already handled with D3 drawing code. 

PR also removes pointer interactions with the legend and address a bit of cleanup.

### CODE REVIEW

**Comment by seve** [comment]
#1707

File: client/src/components/continuousLegend/index.js [hunk 1]
```diff
@@ -110,24 +110,7 @@ const continuous = (selectorId, colorscale, colorAccessor) => {
   colors: state.colors,
 }))
 class ContinuousLegend extends React.Component {
```
Comment (seve) [inline client/src/components/continuousLegend/index.js:112]: Maybe this now calls for PureComponent?

File: client/src/components/continuousLegend/index.js [hunk 2]
```diff
@@ -11,20 +11,20 @@ import {
 
 // create continuous color legend
 // http://bl.ocks.org/syntagmatic/e8ccca52559796be775553b467593a9f
-const continuous = (selectorId, colorscale, colorAccessor) => {
-  const legendheight = 200;
-  const legendwidth = 80;
+const continuous = (selectorId, colorScale, colorAccessor) => {
+  const legendHeight = 200;
```
Comment (tihuan) [inline client/src/components/continuousLegend/index.js:15]: thanks for cleaning up the variable names!

File: client/src/components/continuousLegend/index.js [hunk 3]
```diff
@@ -110,24 +110,7 @@ const continuous = (selectorId, colorscale, colorAccessor) => {
...
-  componentDidMount() {
-    this.updateState(null);
-  }
-
-  componentDidUpdate(prevProps) {
-    this.updateState(prevProps);
-  }
-
-  async updateState(prevProps) {
+  async componentDidUpdate(prevProps) {
```
Comment (tihuan) [inline client/src/components/continuousLegend/index.js:113]: I see that we no longer fetch the colors on DidMount, and only rely on DidUpdate. Does that mean we don't need to render the legend between DidMount and the first DidUpdate call?

**Review by tihuan — COMMENTED** [review]
Thanks for fixing the bug, Seve! One quick question otherwise LGTM!

Will probably still be better to wait for @colinmegill to review, since I'm not super familiar with cellxgene code here 😆 Thank you!

File: client/src/components/continuousLegend/index.js (hunk 3 above)
Comment (seve) [reply client/src/components/continuousLegend/index.js:113]: Yes, the legend is one of the first components to be rendered(it's top level in `App.js`), and cellxgene initializes with `colorBy` set to undefined.  We won't need to render via D3 on DidMount

Comment (tihuan) [reply client/src/components/continuousLegend/index.js:113]: Got it! Thank you 😄  SGTM!

**Review by colinmegill — APPROVED** [review]

**Pull request merged**